src/subtype.c

```diff
@@ -961,6 +961,8 @@ static int subtype(jl_value_t *x, jl_value_t *y, jl_stenv_t *e, int param)
                 if (yy) record_var_occurrence(yy, e, param);
                 if (yr) {
                     if (xx) record_var_occurrence(xx, e, param);
+                    if (xx->lb == jl_bottom_type)
+                        return 1;  // allow Bottom <: non-type
                     return subtype(xx->lb, yy->ub, e, 0);
```

Comment:
I would have thought `subtype` should already handle `lhs == Union{}`, instead of needing to add a fast-path here to catch some cases of it just for the egal case?

Reply:
Yeah, it should be ok to change that. I thought there might be some reason we don't have that general fast path, but I'm not sure what it would be.